employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/contract/ContractsPage.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.gwtui.client.pages.contract;
+
+import java.util.Collections;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import elemental2.dom.HTMLButtonElement;
+import elemental2.dom.MouseEvent;
+import elemental2.promise.Promise;
+import org.jboss.errai.databinding.client.components.ListComponent;
+import org.jboss.errai.databinding.client.components.ListContainer;
+import org.jboss.errai.ioc.client.api.ManagedInstance;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.optaweb.employeerostering.gwtui.client.app.spinner.LoadingSpinner;
+import org.optaweb.employeerostering.gwtui.client.common.EventManager;
+import org.optaweb.employeerostering.gwtui.client.common.EventManager.Event;
+import org.optaweb.employeerostering.gwtui.client.common.FailureShownRestCallback;
+import org.optaweb.employeerostering.gwtui.client.common.KiePager;
+import org.optaweb.employeerostering.gwtui.client.common.KieSearchBar;
+import org.optaweb.employeerostering.gwtui.client.notification.NotificationFactory;
+import org.optaweb.employeerostering.gwtui.client.pages.Page;
+import org.optaweb.employeerostering.gwtui.client.tenant.TenantStore;
+import org.optaweb.employeerostering.gwtui.client.util.PromiseUtils;
+import org.optaweb.employeerostering.shared.contract.Contract;
+import org.optaweb.employeerostering.shared.contract.ContractRestServiceBuilder;
+
+@Templated
+public class ContractsPage
```

Comment:
We discussed the formatting of interfaces in the class declaration with @yurloc and I realized it might not be possible to achieve consistency with KIE group style across different IDEs. Please take this one comment as a soft suggestion.